test/e2e/tests/connections/multiple-standard-connections.spec.ts

```diff
@@ -0,0 +1,392 @@
+import FixtureBuilder from '../../fixture-builder';
+import { TestDappSolana } from '../../page-objects/pages/test-dapp-solana';
+import {
+  connectSolanaTestDapp,
+  DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+  account1 as SOLANA_ADDRESS_ONE,
+} from '../../flask/solana-wallet-standard/testHelpers';
+import { Driver } from '../../webdriver/driver';
+
+const EVM_ADDRESS_TWO = '0x09781764c08de8ca82e156bbf156a3ca217c7950';
+
+const SOLANA_ACCOUNT_ONE = `${SolScope.Mainnet}:${SOLANA_ADDRESS_ONE}`;
+
+const SOLANA_PERMISSIONS = {
+  isMultichainOrigin: true,
+  sessionProperties: {},
+  requiredScopes: {},
+  optionalScopes: {
+    [SolScope.Mainnet]: {
+      accounts: [SOLANA_ACCOUNT_ONE],
+    },
+  },
+};
+
+/**
+ * Helper to open a permissions page for a specific app hostname
+ *
+ * @param driver - The driver to use.
+ * @param hostname - The hostname to get the permissions page for.
+ * @returns The permissions page for the given host.
+ */
+async function getPermissionsPageForHost(driver: Driver, hostname: string) {
+  const homepage = new Homepage(driver);
+  await homepage.headerNavbar.openPermissionsPage();
+  const permissionListPage = new PermissionListPage(driver);
+  await permissionListPage.check_pageIsLoaded();
+  await permissionListPage.openPermissionPageForSite(hostname);
+  const sitePermissionPage = new SitePermissionPage(driver);
+  await sitePermissionPage.check_pageIsLoaded(hostname);
+  return sitePermissionPage;
+}
+
+/**
+ * Helper to get a request permissions request object with a caveat.
+ *
+ * @param accounts - The accounts to be requested.
+ * @returns The request permissions request object with the caveat.
+ */
+function getRequestPermissionsRequestObject(accounts: string[] = []): string {
+  const caveats =
+    accounts.length > 0
+      ? {
+          caveats: [
+            {
+              type: 'restrictReturnedAccounts',
+              value: accounts,
+            },
+          ],
+        }
+      : {};
+
+  return JSON.stringify({
+    jsonrpc: '2.0',
+    method: 'wallet_requestPermissions',
+    params: [{ eth_accounts: caveats }],
+  });
+}
+
+/**
+ * Helper to get a request permissions request object with network restrictions.
+ *
+ * @param networks - Array of network IDs to restrict switching to
+ * @returns the wallet_requestPermissions request string
+ */
+function getRestrictedNetworks(networks: string[]): string {
+  const restrictNetworks = {
+    'endowment:permitted-chains': {
+      caveats: [
+        {
+          type: 'restrictNetworkSwitching',
+          value: networks,
+        },
+      ],
+    },
+  };
+
+  return JSON.stringify({
+    jsonrpc: '2.0',
+    method: 'wallet_requestPermissions',
+    params: [restrictNetworks],
+  });
+}
+
+describe('Multiple Standard Dapp Connections', function () {
+  it('should default account selection to already permitted account(s) when `wallet_requestPermissions` is called with no accounts specified', async function () {
```

Comment:
so it turns out this isn't quite right (which is fine). But we should call it out.

```suggestion
  it('should default account selection to already permitted account(s) plus the selected account (if not already permissioned) when `wallet_requestPermissions` is called with no accounts specified', async function () {
```

https://github.com/user-attachments/assets/8ce38ae5-cdda-4e5e-a93d-851558e16c3a